src/services/node-checker.ts

```diff
@@ -0,0 +1,376 @@
+   * @param path  optional. Blockchain's path to send the request to.
+   * @param source optional but encouraged to be provided. Source to compare the node's blockheight against.
+   * @param allowance optional. Allowed Threshold of number of blocks behind from the node.
+   * @returns response object containing the relay response, request output and boolean
+   * assuring whether the node is on sync with the source, or has a block height over 0 in case no source was provided.
+   */
+  async sync(
+    node: Node,
+    data: string,
+    blockchainID: string,
+    aat: PocketAAT,
+    resultKey: string,
+    path?: string,
+    source?: number,
+    allowance = 0
+  ): Promise<NodeCheckResponse<SyncCheck>> {
+    let blockheight: number
+
+    const isSynced = (payload: object, minimumAllowedHeight) => {
+      blockheight = NodeChecker.parseBlockFromPayload(payload, resultKey)
+
+      if (source > 0 && allowance >= 0) {
+        return blockheight >= minimumAllowedHeight
+      }
+      return blockheight > 0
+    }
+
+    const { relayResponse, success } = await this.processCheck(
+      node,
+      data,
+      blockchainID,
+      aat,
+      path,
+      source - allowance,
+      isSynced
+    )
+
+    if (relayResponse instanceof Error) {
+      return { node, check: 'sync-check', success: false, response: relayResponse, output: { blockHeight: 0 } }
+    }
+
+    return {
+      node,
+      check: 'sync-check',
+      success,
+      response: relayResponse.payload,
+      output: { blockHeight: typeof blockheight === 'number' ? blockheight : 0 },
+    }
+  }
+
+  /**
+   * Perfoms archival check on a node. Making sure the node is capable of performing archival-specific relays in the chain.
+   * @param node Node to check.
+   * @param data Payload to send to the blockchain, expected to return a response that will fail on non-archive nodes.
+   * @param blockchainID Blockchain to request data from.
+   * @param aat Pocket Authentication token object.
+   * @param resultKey key field from the relay response that's expected to have a value on failing non-archive nodes
+   * @param comparator value to compare the resultKey against.
+   * @param path optional. Blockchain's path to send the request to.
+   * @param swap optional. Instead of returning success on archival nodes, return success on not archival nodes.
+   * @returns Response object containing the relay response and boolean.
+   * assuring whether the node supports supports archival or not.
+   */
+  async archival(
+    node: Node,
+    data: string,
+    blockchainID: string,
+    aat: PocketAAT,
+    resultKey: string,
+    comparator: string | number,
+    path?: string,
+    swap = false
+  ): Promise<NodeCheckResponse<ArchivalCheck>> {
+    let payloadResponse: object
+
+    const isArchival = (payload: object, comparatorVal: string) => {
+      payloadResponse = payload
+      const result = NodeChecker.parseBlockFromPayload(payload, resultKey).toString()
```

Comment:
If you are we are `eth_getBalance` for the archival check, then we receive a hex balance. But... we are using function called `parseBlockFromPayload` that does the job (because blocks are also just hex numbers), but it should be renamed. Probably `parseHexFromPayload` is a better name.